### ISSUE TITLE
Support for Yeelight Smart LED Light Panels

### ISSUE THREAD

**Issue by Apocrathia** [syssi/esphome-yeelight-ceiling-light#30]
The new Yeelight Smart LED Light Panels (Model YLFWD-012) appear to have been released with the non-local firmware on the controller. However, upon further inspection, the controller is running an ESP-WROOM-32D. What can we do to get this working with ESPhome?

[Please refer to the following post on the ESPhome forums for more details.](https://community.home-assistant.io/t/yeelight-smart-led-light-panels/370049)

**Comment by syssi** [comment]
I don't own the device so I would like to guide you through the complete process to transform your lamp to an esphome device. The most important and required tool is a multimeter. Do you own a multimeter and do you know how to measure voltage & resistance? :-)

**Comment by syssi** [comment]
1. Please provide a high resolution photo of the PCB holding the ESP32 (both sides)
2. We will try to identify the following pins: `3.3V`, `GND`, `RXD`, `TXD`, `GPIO0` + the GPIOs driving the LEDs

- `3.3V` & `GND` is required to power the board while flashing. If the device has a power supply and is powered by a low voltage we could use the power supply to power the board/esp32.
- `GPIO0` is required to boot the ESP32 into the `download mode` to flash new firmware / make a backup of the stock firmware.
- The GPIOs driving the LEDs: Could you describe the features of the stock firmware? Do you know any details about the LEDs? Could you provide a photo of the LEDs + the connector to attach the LEDs?

**Comment by Apocrathia** [comment]
I've already got all of the necessary tools. Here are all of the pictures that I've taken of the board and accessories.
![IMG_4626](https://user-images.githubusercontent.com/1345711/147370731-7c6a1ad3-c7d3-4d7f-adc2-0b748024c59a.jpeg)
![IMG_4625](https://user-images.githubusercontent.com/1345711/147370733-d984a9a9-cd78-4649-92ee-eabd48ec4d4a.jpeg)
![IMG_4624](https://user-images.githubusercontent.com/1345711/147370735-2c55d3c2-bd26-4d79-9250-25f68928de97.jpeg)
![IMG_4623](https://user-images.githubusercontent.com/1345711/147370737-500ceafa-1d9a-4d63-9179-09ca48fe1aac.jpeg)
![IMG_4622](https://user-images.githubusercontent.com/1345711/147370739-d985cbdf-f4c6-4423-9b6c-8f4a7a5f45bc.jpeg)
![IMG_4621](https://user-images.githubusercontent.com/1345711/147370741-8d3acdda-ccc9-4259-8922-ac5e9ad2a15c.jpeg)
![IMG_4620](https://user-images.githubusercontent.com/1345711/147370742-25e83709-228f-4653-a621-3147dd6dea7a.jpeg)
![IMG_4619](https://user-images.githubusercontent.com/1345711/147370745-8082657c-72a3-42f5-b40d-58473f85bc42.jpeg)
![IMG_4618](https://user-images.githubusercontent.com/1345711/147370747-e12eac58-b924-4573-9671-705a3046f131.jpeg)
![IMG_4617](https://user-images.githubusercontent.com/1345711/147370748-dd869fa5-b784-45f4-8e3b-af9be949e896.jpeg)
![IMG_4616](https://user-images.githubusercontent.com/1345711/147370749-c02acb5f-fcdc-4f73-9aa8-9ceaeb9f82aa.jpeg)
![IMG_4615](https://user-images.githubusercontent.com/1345711/147370751-4f89f7f4-d198-4028-84f6-d5fa0bbc83a4.jpeg)
![IMG_4614](https://user-images.githubusercontent.com/1345711/147370754-747d0cb2-6249-4987-974d-ee52576e51b1.jpeg)
Additionally, here's a reference for the pinout for the ESP-WROOM-32D
![espressif_espwroom32_details](https://user-images.githubusercontent.com/1345711/147370772-b419bb19-151a-4713-a5a2-bebe4180c8ae.jpeg)
It looks like the tx/rx pins are already identified on the board, but I might have to solder leads directly to the ESP for the others.
As of right now, it looks like I am able to get them into HA via the Yeelight integration. It took some digging in the app, but it seems that local control is still available to be enabled. However, I also assume that there will be others purchasing these same lights, and wanting them to work with ESPhome.

**Comment by imhotep** [comment]
@Apocrathia Did you ever get it to work with ESPHome?

**Comment by Apocrathia** [comment]
No, sorry, I haven't really had a chance to work with these. I still haven't even put them up anywhere. I was able to get them pulled into Home Assistant via the Yeelight integration, but I can only control static colors, brightness, and power. None of the effects or audio functions. Because of the number of features provided by the controller, [I've also opened an issue on the python-yeelight repository](https://gitlab.com/stavros/python-yeelight/-/issues/70), as rimplementing everything in ESPHome may be a much larger endeavor than initially expected. However, I simply haven't had an opportunity to really look at it.

**Comment by k37chup** [comment]
Same boat hoping that maybe there was some integration esphome or tasmota but haven't seen anything. I was able to get my hands on a 2nd control board because the first one that was sent to me wasn't broadcasting SSID and I think looking at the chip maybe its due to some overlapping solder joints on the ESP32. Bought some replacements and was hoping to flash something other than the Stock Firmware on there. Anyone have the opportunity to look at these yet?
